### DISCUSSION TITLE
Atinternet exempt et optin / optout

### DISCUSSION THREAD

**Discussion by pierretux** [AmauriC/tarteaucitron.js#918] [General]
Bonjour,

Sur mes sites on me demande d'avoir `tag.privacy.setVisitorMode('cnil', 'exempt');` quand le visiteur arrive sur le site mais de passer `tag.privacy.setVisitorOptin(); ou tag.privacy.setVisitorOptout();` suivant ce que le visiteur fait sur le bandeau de cookie.

Comment récupéré cette info en html ou en php ?

Merci

**Comment by AmauriC** [comment]
C'est géré par le service Atinternet de tarteaucitron ; )

**Comment by pierretux** [comment]
il me renvoi tjs exempt mais si je le mets pas ça remonte pas du coup, comment faire

**Reply by AmauriC** [reply]
Il doit y avoir en plus de tarteaucitron un tag atinternet sur le site. Possible d'avoir l'url du site?

**Reply by pierretux** [reply]
https://solidarites-sante.gouv.fr/

**Reply by AmauriC** [reply]
Ok, merci. Tout me parait ok (excempt en arrivant, optin si acceptation).
Le mode exempt si l'utilisateur refuse est ok mais si vous voulez forcer le mode optout depuis le panel tarteaucitron, c'est possible en ajoutant cet event:
```
document.addEventListener('atinternet_disallowed', function() {window.tag.privacy.setVisitorOptout();});
```

Il faudra par contre mettre à jour tarteaucitron, la version actuelle n'a pas ces nouveaux events.

**Reply by pierretux** [reply]
ok je vais voir cela, ATinternet m'a filer une doc aussi que je vais eplucher pour voir.

Merci

**Reply by AmauriC** [reply]
Une autre chose qui est possible c'est d'ajouter un bouton pour l'optout sur la page "cookie" par exemple.
Quand l'utilisateur refusera sur tarteaucitron, au lieu d'être en excempt ce sera en optout.

**Reply by pierretux** [reply]
oui en bas de https://solidarites-sante.gouv.fr/ministere/article/donnees-personnelles-et-cookies sauf que ce que j'ai ne convient pas manifestement

du coup je test un code comme ça
`<script type="text/javascript">
	function ActivateOptOut() {
		tag.privacy.setVisitorOptout();
		tag.privacy.getVisitorMode();
	}
	function DesactivateOptOut() {
		tag.privacy.setVisitorMode('cnil', 'exempt');
		tag.privacy.getVisitorMode();
	}
</script>
<p><button onclick="ActivateOptOut()">Opt Out Activation</button></p>
<p><button onclick="DesactivateOptOut()">Disable Opt Out</button></p>`

**Reply by pierretux** [reply]
^

> Ok, merci. Tout me parait ok (excempt en arrivant, optin si acceptation). Le mode exempt si l'utilisateur refuse est ok mais si vous voulez forcer le mode optout depuis le panel tarteaucitron, c'est possible en ajoutant cet event:
> 
> ```
> document.addEventListener('atinternet_disallowed', function() {window.tag.privacy.setVisitorOptout();});
> ```
> 
> Il faudra par contre mettre à jour tarteaucitron, la version actuelle n'a pas ces nouveaux events.

@AmauriC quand je navigue sur le site je ne vois pas de exempt dans le tag inpector sur chorme, tu le vois comment ?

**Reply by AmauriC** [reply]
Dans la console Javascript `window.tag.privacy.getVisitorMode();` retourne la valeur actuelle du privacy mode.

**Reply by pierretux** [reply]
je vois pas :( https://paste.pics/2137a0801a29aa672461ee828437d487

**Reply by AmauriC** [reply]
Juste en dessous au niveau des >> bleu + Entrée 🙃

**Comment by pierretux** [comment]
Bon j'ai bien avancé mais quand le visiteur change d'avis cela ne change rien, comment le géré ?

**Reply by pierretux** [reply]
petit up ?

**Reply by AmauriC** [reply]
Change d'avis sur le panel tac?

**Reply by pierretux** [reply]
oui

**Reply by AmauriC** [reply]
![Capture d’écran 2022-04-11 à 15 27 38](https://user-images.githubusercontent.com/8764074/162750322-049f6391-8f5a-465c-865d-80a90606019e.png)

**Reply by pierretux** [reply]
et si on excepte en premier puis un refus on doit avoir quoi ? exempt a la fin ?

**Reply by AmauriC** [reply]
Tarteaucitron mets par défaut exempt (refus ou pas d'acceptation) et passe en optin si l'utilisateur accepte.

Pour gérer le full optout, il faut le faire ailleurs mais si c'est en optout, tarteaucitron le respecte et l'applique au lieu de l'exempt.

**Reply by pierretux** [reply]
je vois pas ou le faire du coup pour que ce soit pris en compte justement
